File: beekeeping-app/src/app/_models/nest-expansion.ts
export class NestExpansion {
    id: number;
    date: Date;
    frameType: FrameType;
    combSheets: number;
    combs: number;
    beefamilyId: number;
}

export enum FrameType {
    NestFrame, HoneyFrame
}

export const FrameType2LabelMapping: Record<FrameType, string> = {
    [FrameType.NestFrame]: "Dideli rėmai",
    [FrameType.HoneyFrame]: "Pusrėmiai"
};

File: beekeeping-api/BeekeepingApi/Models/NestExpansion.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace BeekeepingApi.Models
{
    public enum FrameType
    {
        NestFrame, HoneyFrame
    }

    public class NestExpansion
    {
        [Key]
        public long Id { get; set; }

        public DateTime Date { get; set; }

        public FrameType FrameType { get; set; }

        [Range(0, int.MaxValue)]
        public int CombSheets { get; set; }

        [Range(0, int.MaxValue)]
        public int Combs { get; set; }
        
        public long BeefamilyId { get; set; }

        [ForeignKey("BeefamilyId")]
        public BeeFamily Beefamily { get; set; }
    }
}
